### PR TITLE
Use try-with-resources Statement when is possible.

### PR DESCRIPTION
**Pull request by arturobernalg**
## Impact

- [ ] Bug fix (non-breaking change which fixes expected existing functionality)
- [X ] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Description

A clear and concise description of the change being made.  

Use The try-with-resources Statement when is possible.

### CODE REVIEW

**Comment by nvoxland** [comment]
**Pre-Review Notes**

Like description says, it is cleaning up some try/catch/close with try-with-resources. Doesn't change functionality, just fixes code to use the "new" syntax.

Questions I have:
- None

Potential risks:
- None

What could make the full review difficult:
- Nothing

**Review by MalloD12 — APPROVED** [review]
Approved.

Review and testing results:

Changes looks good to me. Only little changes has been made:
- Merge conflicts resolved.
- Added a catch exception block to capture rest of exceptions

As this is a refactoring with no impact on the logic, since build run successfully with all test passing, I consider this is ready to go.

Things to be aware of:
- None. 

Things to worry about:
- None

File: liquibase-core/src/main/java/liquibase/snapshot/jvm/ColumnSnapshotGenerator.java [hunk 1]
```diff
@@ -276,9 +278,71 @@ protected Column readColumn(CachedRow columnMetadataResultSet, Relation table, D
             }
         }
 
-        if (database.supportsAutoIncrement() && table instanceof Table) {
-            column = this.columnAutoIncrementService.enableColumnAutoIncrementIfAvailable(column, database,
-                    columnMetadataResultSet, rawCatalogName, rawSchemaName, rawTableName, rawColumnName);
+        if (database.supportsAutoIncrement()) {
```
Comment (filipelautert) [inline liquibase-core/src/main/java/liquibase/snapshot/jvm/ColumnSnapshotGenerator.java:281]: Hey, seems that something else was carried over this PR. This is not a try with resources merge but seems a fix from other PR that got merged here...

**Review by filipelautert — CHANGES_REQUESTED** [review]

File: liquibase-core/src/main/java/liquibase/snapshot/jvm/ColumnSnapshotGenerator.java (hunk 1 above)
Comment (MalloD12) [reply liquibase-core/src/main/java/liquibase/snapshot/jvm/ColumnSnapshotGenerator.java:281]: Yeap, this is the result of merge conflict resolved I got when merging master into this branch.

**Review by suryaaki2 — APPROVED** [review]

**Review by filipelautert — APPROVED** [review]

**Pull request merged**